### ISSUE TITLE
Auto Fix on Save does not fix unused disable directives unless there are other lint problems

### ISSUE THREAD

**Issue by shawnmcknight** [microsoft/vscode-eslint#1585]
In my eslint config, I've got the following activated:
```
reportUnusedDisableDirectives: true,
```

This causes eslint to emit a warning for unused disable directives and it properly shows as a warning in vscode:
![image](https://user-images.githubusercontent.com/4583869/213741753-fcfd6859-c6a1-4cb9-b260-15b4f4b4ac3e.png)

In my vscode settings I've got code actions on save enabled:
```
"editor.codeActionsOnSave": {
  "source.fixAll": true
},
```

However, upon save, this warning is not autofixed.  Additionally, there is no "quick fix" shown in vscode as there would be for other fixable errors.

This warning is autofixable at command line by eslint using the `--fix` flag.

I have found that if there are other errors in the file then upon save both the other error as well as the unused directive will be autofixed.  Additionally, if I choose the quick fix "Fix all auto-fixable problems" for the other error, it will automatically fix the unused directive.  It appears that automatic fixing will only not occur if this is the only problem in the file.

It's possible I'm missing a configuration somewhere that would allow this to be fixed and if that is the case would appreciate guidance on how to configure it.  If that is not the case then please consider a feature request to the extension to treat this as a fixable error.

Thanks!

**Comment by MariaSolOs** [comment]
I debugged this a bit and the issue is that ESLint does not assign a `ruleId` to disable directive reports, and this extension uses that ID for generating items for the "quick fix" menu:
https://github.com/microsoft/vscode-eslint/blob/d34c6b23529e4136fe75fb591deedeb3c270ca87/server/src/eslint.ts#L1108-L1115 

I think we could have a special treatment of these errors to enable the requested functionality...

**Comment by dbaeumer** [comment]
@MariaSolOs thanks for debugging this. I agree that we should have special treatment for this. Are you willing to provide a PR?

**Comment by MariaSolOs** [comment]
> @MariaSolOs thanks for debugging this. I agree that we should have special treatment for this. Are you willing to provide a PR?

I would be happy to :)
